src/utils/openai.ts

```diff
@@ -100,18 +97,45 @@ const sanitizeMessage = (message: string) => message.trim().replace(/[\n\r]/g, '
 
 const deduplicateMessages = (array: string[]) => Array.from(new Set(array));
 
-const getPrompt = (locale: string, diff: string) => `Write a git commit message in present tense for the following diff without prefacing it with anything. Do not be needlessly verbose and make sure the answer is concise and to the point. The response must be in the language ${locale}:\n${diff}`;
+const getPrompt = (locale: string, diff: string, length: number) => `Write a git commit message in present tense for the following diff without prefacing it with anything. Do not be needlessly verbose and make sure the answer is concise and to the point. The response must be no longer than ${length} characters. The response must be in the language ${locale}:\n${diff}`;
+
+const generateStringFromLength = (length: number) => {
+	let result = '';
+	const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
+	for (let i = 0; i < length; i += 1) {
+		const randomIndex = Math.floor(Math.random() * characters.length);
+		result += characters.charAt(randomIndex);
+	}
+	return result;
+};
+
+const getTokens = (prompt: string, model: TiktokenModel) => {
+	const encoder = encoding_for_model(model);
+	const tokens = encoder.encode(prompt).length;
+	// Free the encoder to avoid possible memory leaks.
+	encoder.free();
+	return tokens;
+};
 
 export const generateCommitMessage = async (
 	apiKey: string,
 	model: TiktokenModel,
 	locale: string,
 	diff: string,
 	completions: number,
+	length: number,
 	timeout: number,
 	proxy?: string,
 ) => {
-	const prompt = getPrompt(locale, diff);
+	const prompt = getPrompt(locale, diff, length);
+
+	// Padded by 5 for more room for the completion.
+	const stringFromLength = generateStringFromLength(length + 5);
+	const tokenFromLength = getTokens(stringFromLength, model);
+	const tokenFromPrompt = getTokens(prompt, model);
```

Comment:
Is it possible to concatenate `prompt + stringFromLength` and pass it into `getTokens` once?

That way you can remove the `getTokens` util.